src/main/java/se/sundsvall/contract/service/ContractService.java

```diff
@@ -41,12 +51,22 @@ public String createContract(final String municipalityId, final Contract contrac
 	}
 
 	@Transactional(readOnly = true)
-	public Contract getContract(final String municipalityId, final String contractId) {
-		return contractRepository.findFirstByMunicipalityIdAndContractIdOrderByVersionDesc(municipalityId, contractId)
-			.map(contractEntity -> contractMapper.toContractDto(contractEntity, attachmentRepository.findAllByMunicipalityIdAndContractId(municipalityId, contractId)))
+	public Contract getContract(final String municipalityId, final String contractId, final Integer version) {
+		Optional<ContractEntity> contractEntity;
+
+		if (version == null) {
+			contractEntity = contractRepository.findFirstByMunicipalityIdAndContractIdOrderByVersionDesc(municipalityId, contractId);
+		} else {
+			contractEntity = contractRepository.findByMunicipalityIdAndContractIdAndVersion(municipalityId, contractId, version);
+		}
+
+		return contractEntity
```

Comment:
```
return Optional.ofNullable(version)
          .map(v -> contractRepository.findByMunicipalityIdAndContractIdAndVersion(municipalityId, contractId, v)
          .orElseGet(contractRepository.findFirstByMunicipalityIdAndContractIdOrderByVersionDesc(municipalityId, contractId))
          .map(entity -> contractMapper.toContractDto(entity, attachmentRepository.findAllByMunicipalityIdAndContractId(municipalityId, contractId)))
			.orElseThrow(() -> Problem.builder()
				.withStatus(Status.NOT_FOUND)
				.withDetail(version != null ?
					CONTRACT_ID_MUNICIPALITY_ID_VERSION_NOT_FOUND.formatted(contractId, version, municipalityId) :
					CONTRACT_ID_MUNICIPALITY_ID_NOT_FOUND.formatted(contractId, municipalityId))
				.build());

```

One-liners are cooler

Reply:
Your suggestion won't work as Nintendo, since the first `findByMunicipalityIdAndContractIdAndVersion` call for a non-existent entity would mean that the second `findFirstByMunicipalityIdAnd...` call will be executed. If there is a request for a specific version that doesn't exist, not-found-behaviour is desired. If a request is made without specifying the version, the latest should be returned.